modules/pubsub/broker.go

```diff
@@ -0,0 +1,101 @@
+package pubsub
+
+import (
+	"sync"
+
+	"github.com/spaceuptech/space-cloud/model"
+	"github.com/spaceuptech/space-cloud/utils"
+)
+
+natsWildcard string = ">"
+
+type subscription interface {
+	Unsubscribe() error
+}
+
+type pubsubBroker interface {
+	Publish(subject string, msg *model.PubsubMsg) error
+	QueueSubscribe(subject, group string, ch chan *model.PubsubMsg) (*subscription, error)
```

Comment:
There should also be a normal subscribe right?

Reply:
@AliabbasMerchant i suggest you keep a nornal subscribe as well. Internally call QueueSubscribe.

Reply:
This is not exposed to the outside world. This is just for internal use.

Reply:
I understand. But let that logic be inside.

Reply:
Actually its alright